integration/feature/init/src/MillInitSbtTests.scala

```diff
@@ -0,0 +1,198 @@
+      import tester.*
+      bumpSbtTo1107(workspacePath)
+
+      testMillInit(
+        tester,
+        expectedCompileTasks =
+          Some(SplitResolvedTasks(successful = Seq("compile", "test.compile"), failed = Seq.empty)),
+        expectedTestTasks = Some(SplitResolvedTasks(successful = Seq("test"), failed = Seq.empty))
+      )
+    }
+  }
+}
+
+// same as the one in the unit tests
+object MillInitSbtMultiProjectExampleTests extends BuildGenTestSuite {
+  def tests: Tests = Tests {
+    /*
+    - 12 KB
+    - originally sbt 1.0.2
+     */
+    val url =
+      "https://github.com/pbassiner/sbt-multi-project-example/archive/152b31df9837115b183576b0080628b43c505389.zip"
+
+    test - integrationTest(url) { tester =>
+      import tester.*
+      bumpSbtTo1107(workspacePath)
+      /*
+      `multi1.compile` doesn't work well when Mill is run with JDK 17 and 21:
+      ```text
+      1 tasks failed
+      multi1.compile java.io.IOError: java.lang.RuntimeException: /packages cannot be represented as URI
+          java.base/jdk.internal.jrtfs.JrtPath.toUri(JrtPath.java:175)
+          scala.tools.nsc.classpath.JrtClassPath.asURLs(DirectoryClassPath.scala:183)
+          ...
+      ```
+      Passing a `jvmId` 11 doesn't work here.
+       */
+      writeMillJvmVersionTemurin11(workspacePath)
+
+      val submodules = Seq("common", "multi1", "multi2")
+      testMillInit(
+        tester,
+        expectedCompileTasks = Some(SplitResolvedTasks(
+          successful = Seq("compile") ++ submodules.flatMap(allCompileTasks),
+          failed = Seq.empty
+        )),
+        expectedTestTasks =
+          Some(SplitResolvedTasks(successful = submodules.map(testTask), failed = Seq.empty))
+      )
+    }
+  }
+}
+
+// relatively large libraries
+
+object MillInitSbtGatlingTests extends BuildGenTestSuite {
+  def tests: Tests = Tests {
+    /*
+    - 1.8 MB
+    - sbt 1.10.7
+     */
+    val url = "https://github.com/gatling/gatling/archive/refs/tags/v3.13.4.zip"
+
+    val submodules = Seq(
+      "gatling-app",
+      "gatling-benchmarks",
+      "gatling-charts",
+      "gatling-commons",
+      "gatling-core-java",
+      "gatling-core",
+      "gatling-http-client",
+      "gatling-http-java",
+      "gatling-http",
+      "gatling-jdbc-java",
+      "gatling-jdbc",
+      "gatling-jms-java",
+      "gatling-jms",
+      "gatling-jsonpath",
+      "gatling-netty-util",
+      "gatling-quicklens",
+      "gatling-recorder",
+      "gatling-redis-java",
+      "gatling-redis",
+      "gatling-samples",
+      "gatling-test-framework"
+    )
+    val submodulesWithoutTests = Seq(
+      "gatling-app",
+      "gatling-benchmarks",
+      "gatling-quicklens",
+      "gatling-samples",
+      "gatling-test-framework"
+    )
+    val submodulesWithTests = submodules diff submodulesWithoutTests
+
+    test - integrationTest(url) { tester =>
+      testMillInit(
+        tester,
+        expectedCompileTasks = Some(SplitResolvedTasks(
```

Comment:
Could we add assertions on the number of source files in the `allSourceFiles` task, for both the `main` and the `test` submodules? That would help us catch any issues with the source folders being mis-configured, resulting in no source files being passed and thus compile/test passing by default

Reply:
OK I will try.